octodns/provider/mythicbeasts.py

```diff
@@ -0,0 +1,420 @@
+    return value[:-1]
+
+
+class MythicBeastsProvider(BaseProvider):
+    '''
+    Mythic Beasts DNS API Provider
+
+    mythicbeasts:
+      class: octodns.provider.mythicbeasts.MythicBeastsProvider
+        zones:
+          my-zone: 'password'
+    '''
+
+    SUPPORTS_GEO = False
+    SUPPORTS_DYNAMIC = False
+    SUPPORTS = set(('A', 'AAAA', 'ALIAS', 'CNAME', 'MX', 'NS',
+                    'SRV', 'SSHFP', 'CAA', 'TXT'))
+    BASE = 'https://dnsapi.mythic-beasts.com/'
+
+    def __init__(self, identifier, passwords, *args, **kwargs):
+        self.log = getLogger('MythicBeastsProvider[{}]'.format(identifier))
+
+        assert isinstance(passwords, dict), 'Missing passwords'
+
+        self.log.debug(
+            '__init__: id=%s, registered zones; %s',
+            identifier,
+            passwords.keys())
+        super(MythicBeastsProvider, self).__init__(identifier, *args, **kwargs)
+
+        self._passwords = passwords
+        sess = Session()
+        self._sess = sess
+
+    def _request(self, method, path, data=None):
+        self.log.debug('_request: method=%s, path=%s data=%s',
+                       method, path, data)
+
+        resp = self._sess.request(method, path, data=data)
+        self.log.debug(
+            '_request:   status=%d data=%s',
+            resp.status_code,
+            resp.text[:20])
+
+        if resp.status_code == 401:
+            raise Exception('Mythic Beasts unauthorized for domain: {}'
+                            .format(data['domain']))
+        resp.raise_for_status()
+        return resp
+
+    def _post(self, data=None):
+        return self._request('POST', self.BASE, data=data)
+
+    def records(self, zone):
+        assert zone in self._passwords, 'Missing password for domain: {}' \
+            .format(remove_trailing_dot(zone))
+
+        return self._post({
+            'domain': remove_trailing_dot(zone),
+            'password': self._passwords[zone],
+            'showall': 0,
+            'command': 'LIST',
+        })
+
+    @staticmethod
+    def _data_for_single(_type, data):
+        return {
+            'type': _type,
+            'value': data['raw_values'][0]['value'],
+            'ttl': data['raw_values'][0]['ttl']
+        }
+
+    @staticmethod
+    def _data_for_multiple(_type, data):
+        return {
+            'type': _type,
+            'values':
+                [raw_values['value'] for raw_values in data['raw_values']],
+            'ttl':
+                max([raw_values['ttl'] for raw_values in data['raw_values']]),
+        }
+
+    @staticmethod
+    def _data_for_MX(_type, data):
+        ttl = max([raw_values['ttl'] for raw_values in data['raw_values']])
+        values = []
+
+        for raw_value in \
+                [raw_values['value'] for raw_values in data['raw_values']]:
+            match = re.match('^([0-9]+)\\s+(\\S+)$', raw_value, re.IGNORECASE)
+
+            assert match is not None, 'Unable to parse MX data'
```

Comment:
Not 100% sure, but `assert`ing here may get in the way of the `lenient` flag working as designed for this provider. In general that does a best effort building of records so that a baseline set of data can be pulled out of a provider when things aren't 100% right. When you go to sync the resulting yaml things will complain and requiring fixing then (unless you further set `lenient`.)

The functionality is intended to allow users to pull less than ideal data out of a provider, clean it up locally and then push back (and/or to somewhere else) better data.